menus/pregame.py

```diff
@@ -144,19 +162,21 @@ def sustain_mod(self):
             asm.JMP(options_table, asm.ABS_X_16),
 
             "SUSTAIN_SCROLL_AREA",
-            asm.LDA(0x0d, asm.DIR),
+            asm.LDA(0x09, asm.DIR),
             asm.BIT(0x80, asm.IMM8),        # b pressed?
-            asm.BNE("EXIT_SCROLL_AREA"),
+            asm.BNE("EXIT_SCROLL_AREA"),    # branch if so
+
+        ]
+
+        src.extend(self.common.get_flags_a_check_src(self.invoke_flags_submenu[submenu_idx]))
```

Comment:
⛔Currently if we have no submenus defined (i.e. `-rls` is not set) this will error, as submenu_idx is relying on at least one iteration on one level

I did fix this, will add some suggestions to the PR

### Change 1: Iterate over the keys so submenu_id is always defined
```suggestion
        for submenu_idx in self.common.flags.submenus.keys():
            src.extend(self.common.get_flags_a_check_src(submenu_idx, self.invoke_flags_submenu[submenu_idx]))
```

Reply:
Thanks, I've incorporated this change.